meetings/working-groups/collection-literals/CL-2023-08-10.md

```diff
@@ -0,0 +1,66 @@
+# Collection literals working group meeting for August 10, 2023
+
+## Agenda
+
+* Determine behavior when target runtime does not support Inline-Arrays.
+
+## Discussion
+
+The purpose of the meeting was to go over the decisions made in the last WG meeting (https://github.com/dotnet/csharplang/blob/main/meetings/working-groups/collection-literals/CL-2023-08-03.md) and consider their impact in the context of users targeting runtimes which do not support inline-arrays.
+
+As a reminder, the final conclusion from that meeting was:
+
+> Working group goes with option 3. Collection expressions targeted to ref-struct types are always stack allocated.
+
+Currently, the approach we are intending to take for targeting something like `Span<T>/ReadOnlySpan<T>` is to use the [inline-arrays](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/inline-arrays.md) feature to synthesize an inline-array to place the data to, and then obtain a span from it.  However, inline-arrays are only available on Net8.0 and up.
+
+Further discussion on this topic raised concerns from several participants that this approach was too onerous in practice.  Specifically, that a core use case in modern .NET is for users to multi-target their code.  This multi-targeting appears in practice with users targeting frameworks like netstandard2.0 and netX.0, or just multiple versions of netX.0 (like net6.0/net7.0/net8.0/etc.). This raises a dilemma where users who are multi-targeting are then shut out from being able to use collection expressions with spans.
+
+However, it is possible for the compiler to still generate code in these cases.  Specifically, by allocating an array (on the heap) and store it in the span.  Note: this would not change the local-versus-global scope aspect of the span.  The decision on if the span had local or global scope would use the same rules decided from the last working group.  It would just be the case that a local-scope'd span would be stack-allocated on targets that supported inline-arrays, but heap allocated on targets that did not.  This is necessary so that a user moving from a prior runtime (without inline-array support) to a current runtime would not find their scopes become more restrictive, certainly breaking user code that had come to depend on the global scope from before.
```

Comment:
```suggestion
Further discussion on this topic raised concerns from several participants that this approach was too onerous in practice.  Specifically, that a core use case in modern .NET is for users to multi-target their code.  This multi-targeting appears in practice with users targeting frameworks like netstandard2.0 and netX.0, or just multiple versions of netX.0 (like net6.0/net7.0/net8.0/etc.). This raises a dilemma where users who are multi-targeting are then shut out from being able to use collection expressions with spans.

Stack allocating, though, is not a mandatory requirement. It is possible for the compiler to still generate code in these cases.  Specifically, by allocating an array (on the heap) and store it in the span.  Note: this would not change the local-versus-global scope aspect of the span.  The decision on if the span had local or global scope would use the same rules decided from the last working group.  It would just be the case that a local-scope'd span would be stack-allocated on targets that supported inline-arrays, but heap allocated on targets that did not.  This is necessary so that a user moving from a prior runtime (without inline-array support) to a current runtime would not find their scopes become more restrictive, certainly breaking user code that had come to depend on the global scope from before.
```